fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/VectorIndexHelper.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * VectorIndexHelper.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.indexes;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.async.hnsw.Config;
+import com.apple.foundationdb.async.hnsw.Config.ConfigBuilder;
+import com.apple.foundationdb.async.hnsw.HNSW;
+import com.apple.foundationdb.linear.Metric;
+import com.apple.foundationdb.record.logging.LogMessageKeys;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.IndexOptions;
+import com.apple.foundationdb.record.metadata.MetaDataException;
+import com.apple.foundationdb.record.provider.common.StoreTimer;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Helper functions for index maintainers that use a {@link HNSW}.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class VectorIndexHelper {
```

Comment:
Teamscale wants this to be `final` as there is no public constructor. Which is fair enough, though I might ask whether this should instead be package-private, or if there are reasons we'd expect external code to need to access this class. Is it just the `Event`s? If so, maybe those should be moved out